packages/server/src/app/db/sections/sections.module.ts

```diff
@@ -0,0 +1,48 @@
+import { Module } from '@nestjs/common';
+import { MongooseModule } from '@nestjs/mongoose';
+import { HookNextFunction } from 'mongoose';
+import { generate } from 'shortid';
+
+import { sanitizeHtml, stripAllHtml } from '@pulp-fiction/html_sanitizer';
+import { SectionsDocument, SectionsSchema } from './sections.schema';
+import { SectionsService } from './sections.service';
+import { countPlaintextWords, countQuillWords } from '@pulp-fiction/word_counter';
+
+@Module({
+  imports: [
+    MongooseModule.forFeatureAsync([
+      {
+        name: 'Section',
+        useFactory: () => {
+          const schema = SectionsSchema;
+          schema.pre<SectionsDocument>('save', async function (next: HookNextFunction) {
+            this.set('_id', generate());
+            this.set('title', await sanitizeHtml(this.title));
+            this.set('body', await sanitizeHtml(this.body));
+            if (this.authorsNote) {
+              this.set('authorsNote', await sanitizeHtml(this.authorsNote));
+            }
+            if (this.authorsNotePos) {
+              this.set('authorsNotePos', this.authorsNotePos);
+            }
+            this.set('published', this.published);
+
+            const wordCount = this.usesNewEditor
+              ? await countPlaintextWords(await stripAllHtml(this.body))
+              : await countQuillWords(await sanitizeHtml(this.body));
+            this.set('stats.words', wordCount);
+
+            this.set('createdAt', Date.now());
+            this.set('updatedAt', Date.now());
```

Comment:
so I ran into this when doing notifications--this totally breaks these fields--any time this document gets updated, both `createdAt` and `updatedAt` get set to the current date. that's fine for `updatedAt`. `createdAt`, not so much.

_HOWEVER_ letting Mongoose do its thing seemed to work just fine for notifications. Check to see if you can do the same here.